### PR TITLE
Lots of automated migrations to Symfony Flex structure

### PR DESCRIPTION
**Pull request by wouterj**
Almost all of these changes need a check if the paragraphs surrounding the code block still make sense, but at least, it is a start.

I skipped the Best Pratices and Create Your Own Framework guides. Both of these need to be completely revisited.

Also, lots of other (configuration, bundles, controller and services) guides need to be checked carefully.

Please review and merge quickly @symfony/team-symfony-docs

### CODE REVIEW

File: components/routing.rst [hunk 1]
```diff
@@ -364,8 +364,7 @@ routes with UTF-8 characters:
 
     .. code-block:: php-annotations
 
-        // src/AppBundle/Controller/DefaultController.php
```
Comment (fabpot) [inline components/routing.rst]: Would it be possible to keep such comments? We can probably automatically just strip the AppBundle directory. The new one would be `src/Controller/DefaultController.php` in that case.

Comment (wouterj) [reply components/routing.rst]: Yes, I changed it everywhere, but we're in the components section, we shouldn't talk about framework conventions here. (probably, this needs to be extracted to its own subguide).

Comment (fabpot) [reply components/routing.rst]: ok, makes sense.

File: form/form_dependencies.rst [hunk 2]
```diff
@@ -105,23 +105,23 @@ manually and tag it with ``form.type``:
 
     .. code-block:: yaml
 
-        # src/AppBundle/Resources/config/services.yml
+        # src/Resources/config/services.yaml
```
Comment (yceruto) [inline form/form_dependencies.rst]: It should refer to `config/services.yaml`? same for other formats.

File: http_cache.rst [hunk 3]
```diff
@@ -82,17 +82,17 @@ that wraps the default one (``AppKernel``). The caching Kernel *is* the reverse
...
+changes to ``index.php`` to add caching to the ``dev`` environment::
 
-    // web/app.php
+    // public/index.php
     use Symfony\Component\HttpFoundation\Request;
 
     // ...
-    $kernel = new AppKernel('prod', false);
+    $kernel = new Kernel('prod', false);
     $kernel->loadClassCache();
```
Comment (yceruto) [inline http_cache.rst]: this line is obsolete?

File: controller/argument_value_resolver.rst [hunk 4]
```diff
@@ -148,14 +148,14 @@ and adding a priority.
 
     .. code-block:: yaml
 
-        # app/config/services.yml
+        # app/config/services.yaml
```
Comment (yceruto) [inline controller/argument_value_resolver.rst]: `config/*` everywhere?

File: deployment/azure-website.rst [hunk 5]
```diff
@@ -294,7 +294,7 @@ below:
 
 The code of the Symfony application has now been deployed to the Azure Website
 which you can browse from the file explorer of the Kudu application. You should
-see the ``app/``, ``src/`` and ``web/`` directories under your ``site/wwwroot``
+see the ``app/``, ``src/`` and ``public/`` directories under your ``site/wwwroot``
```
Comment (yceruto) [inline deployment/azure-website.rst]: `"see the ``config/``, ..."`

File: http_cache/esi.rst [hunk 6]
```diff
@@ -126,7 +126,7 @@ matter), Symfony uses the standard ``render`` helper to configure ESI tags:
 
     .. code-block:: twig
 
-        {# app/Resources/views/static/about.html.twig #}
+        {# templates/static/about.html.twig #}
 
         {# you can use a controller reference #}
         {{ render_esi(controller('AppBundle:News:latest', { 'maxPerPage': 5 })) }}
```
Comment (yceruto) [inline http_cache/esi.rst]: `App\Controller\NewsController::latestAction` use FQNC notation everywhere?

File: security/csrf_in_login_form.rst [hunk 7]
```diff
@@ -132,7 +132,7 @@ using the login form:
 
     .. code-block:: html+twig
 
-        {# src/AppBundle/Resources/views/Security/login.html.twig #}
+        {# src/Resources/views/Security/login.html.twig #}
```
Comment (yceruto) [inline security/csrf_in_login_form.rst]: maybe this templates should be moved to `templates/`?

File: security/custom_authentication_provider.rst [hunk 8]
```diff
@@ -459,10 +459,10 @@ to service ids that may not exist yet: ``AppBundle\Security\Authentication\Provi
 Now that your services are defined, tell your security context about your
 factory in your bundle class::
 
-    // src/AppBundle/AppBundle.php
+    // src/AppBundle.php
     namespace AppBundle;
```
Comment (yceruto) [inline security/custom_authentication_provider.rst]: both lines should be updated, probably this code lives now in `Kernel.php`.

File: bundles/override.rst [hunk 9]
```diff
@@ -26,7 +26,7 @@ Routing
 
 Routing is never automatically imported in Symfony. If you want to include
 the routes from any bundle, then they must be manually imported from somewhere
-in your application (e.g. ``app/config/routing.yml``).
+in your application (e.g. ``app/config/routes.yaml``).
```
[the remark this answers is not in the document]
Comment (wouterj) [reply bundles/override.rst]: Reverted this change, the whole paragraph needs to be rewritten (and thus done in a non-automatic PR).

File: console.rst [hunk 10]
```diff
@@ -229,9 +227,9 @@ class. It uses special input and output classes to ease testing without a real
 console::
 
     // tests/AppBundle/Command/CreateUserCommandTest.php
-    namespace Tests\AppBundle\Command;
+    namespace Tests\App\Command;
```
[the remark this answers is not in the document]
Comment (wouterj) [reply console.rst]: Fixed (every other occurence of this as well)

File: controller/argument_value_resolver.rst (hunk 4 above)
Comment (wouterj) [reply controller/argument_value_resolver.rst]: Done for all services and routes files, the remaining onces need all be visited manually.

File: http_cache.rst (hunk 3 above)
Comment (wouterj) [reply http_cache.rst]: Reverted, all front controller and kernel changes should be done manually.

File: http_cache/esi.rst (hunk 6 above)
Comment (wouterj) [reply http_cache/esi.rst]: Let's change that in a manual PR as well (not so easy to automate)

**Comment by wouterj** [comment]
Thanks a lot for your detailed review @yceruto ! I've now fixed everything or reverted the change as they have to be done in another PR (to not make this too big and hard to merge).

File: service_container.rst [hunk 11]
```diff
@@ -1067,5 +1067,5 @@ Learn more
     /service_container/*
 
 .. _`service-oriented architecture`: https://en.wikipedia.org/wiki/Service-oriented_architecture
-.. _`Symfony Standard Edition (version 3.3) services.yaml`: https://github.com/symfony/symfony-standard/blob/3.3/app/config/services.yaml
+.. _`Symfony Standard Edition (version 3.3) services.yaml`: https://github.com/symfony/symfony-standard/blob/3.3/config/services.yaml
```
Comment (yceruto) [inline service_container.rst]: should be reverted or referenced to the recipe instead?

**Review by yceruto — COMMENTED** [review]
Awesome!

File: service_container.rst (hunk 11 above)
Comment (wouterj) [reply service_container.rst]: Agreed

**Comment by wouterj** [comment]
Applied the comment and also fixed the build

File: security/custom_authentication_provider.rst [hunk 12]
```diff
@@ -457,24 +457,23 @@ to service ids that may not exist yet: ``AppBundle\Security\Authentication\Provi
...
+factory in the kernel::
 
-    // src/AppBundle.php
-    namespace AppBundle;
+    // src/Kernel.php
+    namespace App;
 
     use App\DependencyInjection\Security\Factory\WsseFactory;
-    use Symfony\Component\HttpKernel\Bundle\Bundle;
-    use Symfony\Component\DependencyInjection\ContainerBuilder;
```
Comment (yceruto) [inline security/custom_authentication_provider.rst]: `use Symfony\Component\HttpKernel\Kernel as BaseKernel`?

**Comment by weaverryan** [comment]
YES, awesome! Thank you Wouter! Thanks to the review from @yceruto, I'm happy to merge this quickly. If we find any bugs, we can make sure to automate those "fixes" across everything. So I think merging this right now is a great idea :)

**Pull request merged**